### ISSUE TITLE
accuracy on log and test.py are quite different.

### ISSUE THREAD

**Issue by Ultraman82** [open-mmlab/mmaction2#422]
Running with TSM. on the log, accuracy is quite high.

```bash
2020-12-07 11:20:41,295 - mmaction - INFO - Epoch [100][34/36]	lr: 5.000e-05, eta: 0:00:02, time: 0.452, data_time: 0.001, memory: 8734, top1_acc: 0.9000, top3_acc: 0.9750, loss_cls: 0.2078, loss: 0.2078, grad_norm: 5.4866
2020-12-07 11:20:41,746 - mmaction - INFO - Epoch [100][35/36]	lr: 5.000e-05, eta: 0:00:01, time: 0.452, data_time: 0.001, memory: 8734, top1_acc: 0.7750, top3_acc: 0.9750, loss_cls: 0.4313, loss: 0.4313, grad_norm: 7.9023
2020-12-07 11:20:42,111 - mmaction - INFO - Epoch [100][36/36]	lr: 5.000e-05, eta: 0:00:00, time: 0.364, data_time: 0.001, memory: 8734, top1_acc: 0.8750, top3_acc: 0.9688, loss_cls: 0.3569, loss: 0.3569, grad_norm: 8.2815 

but testing with result, shows like this.

top1_acc: 0.4764
top3_acc: 0.7406
```

the result with test.py seems quite right. 
I wonder how to change log accuracy to actual accuracy. or wonder what I am doing wrong.

```yaml
model = dict(
    type='Recognizer2D',
    backbone=dict(
        type='ResNetTSM',
        pretrained='torchvision://resnet50',
        depth=50,
        norm_eval=False,
        shift_div=8),
    cls_head=dict(
        type='TSMHead',
        num_classes=21,
        in_channels=2048,
        spatial_type='avg',
        consensus=dict(type='AvgConsensus', dim=1),
        dropout_ratio=0.5,
        init_std=0.001,
        is_shift=True))
train_cfg = None
test_cfg = dict(average_clips='prob')
dataset_type = 'RawframeDataset'
data_root = '/home/oem/data/catdog/'
data_root_val = '/home/oem/data/catdog/'
ann_file_train = '/home/oem/data/catdog/train2608_nocap.txt'
ann_file_val = '/home/oem/data/catdog/val2608_nocap.txt'
ann_file_test = '/home/oem/data/catdog/test2608_nocap.txt'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_bgr=False)
train_pipeline = [
    dict(type='SampleFrames', clip_len=1, frame_interval=1, num_clips=8),
    dict(type='RawFrameDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(
        type='MultiScaleCrop',
        input_size=224,
        scales=(1, 0.875, 0.75, 0.66),
        random_crop=False,
        max_wh_scale_gap=1,
        num_fixed_crops=13),
    dict(type='Resize', scale=(224, 224), keep_ratio=False),
    dict(type='Flip', flip_ratio=0.5),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_bgr=False),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs', 'label'])
]
val_pipeline = [
    dict(
        type='SampleFrames',
        clip_len=1,
        frame_interval=1,
        num_clips=8,
        test_mode=True),
    dict(type='RawFrameDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(type='CenterCrop', crop_size=224),
    dict(type='Flip', flip_ratio=0),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_bgr=False),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]
test_pipeline = [
    dict(
        type='SampleFrames',
        clip_len=1,
        frame_interval=1,
        num_clips=8,
        test_mode=True),
    dict(type='RawFrameDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(type='CenterCrop', crop_size=224),
    dict(type='Flip', flip_ratio=0),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_bgr=False),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]
data = dict(
    videos_per_gpu=10,
    workers_per_gpu=5,
    train=dict(
        type='RawframeDataset',
        ann_file='/home/oem/data/catdog/train2608_nocap.txt',
        data_prefix='/home/oem/data/catdog/',
        pipeline=[
            dict(
                type='SampleFrames', clip_len=1, frame_interval=1,
                num_clips=8),
            dict(type='RawFrameDecode'),
            dict(type='Resize', scale=(-1, 256)),
            dict(
                type='MultiScaleCrop',
                input_size=224,
                scales=(1, 0.875, 0.75, 0.66),
                random_crop=False,
                max_wh_scale_gap=1,
                num_fixed_crops=13),
            dict(type='Resize', scale=(224, 224), keep_ratio=False),
            dict(type='Flip', flip_ratio=0.5),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_bgr=False),
            dict(type='FormatShape', input_format='NCHW'),
            dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
            dict(type='ToTensor', keys=['imgs', 'label'])
        ]),
    val=dict(
        type='RawframeDataset',
        ann_file='/home/oem/data/catdog/val2608_nocap.txt',
        data_prefix='/home/oem/data/catdog/',
        pipeline=[
            dict(
                type='SampleFrames',
                clip_len=1,
                frame_interval=1,
                num_clips=8,
                test_mode=True),
            dict(type='RawFrameDecode'),
            dict(type='Resize', scale=(-1, 256)),
            dict(type='CenterCrop', crop_size=224),
            dict(type='Flip', flip_ratio=0),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_bgr=False),
            dict(type='FormatShape', input_format='NCHW'),
            dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
            dict(type='ToTensor', keys=['imgs'])
        ]),
    test=dict(
        type='RawframeDataset',
        ann_file='/home/oem/data/catdog/test2608_nocap.txt',
        data_prefix='/home/oem/data/catdog/',
        pipeline=[
            dict(
                type='SampleFrames',
                clip_len=1,
                frame_interval=1,
                num_clips=8,
                test_mode=True),
            dict(type='RawFrameDecode'),
            dict(type='Resize', scale=(-1, 256)),
            dict(type='CenterCrop', crop_size=224),
            dict(type='Flip', flip_ratio=0),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_bgr=False),
            dict(type='FormatShape', input_format='NCHW'),
            dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
            dict(type='ToTensor', keys=['imgs'])
        ]))
optimizer = dict(
    type='SGD',
    constructor='TSMOptimizerConstructor',
    paramwise_cfg=dict(fc_lr5=True),
    lr=0.005,
    momentum=0.9,
    weight_decay=0.0001)
optimizer_config = dict(grad_clip=dict(max_norm=20, norm_type=2))
lr_config = dict(policy='step', step=[40, 80])
total_epochs = 100
checkpoint_config = dict(interval=5)
evaluation = dict(
    interval=10, metrics=['top_k_accuracy', 'mean_class_accuracy'])
log_config = dict(interval=1, hooks=[dict(type='TextLoggerHook')])
dist_params = dict(backend='nccl')
log_level = 'INFO'
work_dir = './work_dirs/2608_tsm_r50/'
load_from = None
resume_from = './work_dirs/2608_tsm_r50/epoch_70.pth'
workflow = [('train', 1)]
gpu_ids = range(0, 1)
omnisource = False

```

**Comment by kennymckormick** [comment]
The reason is that: the log displays the accuracy on the training set while testing displays the accuracy on the validation set. To see validation acc during training, you should train with `--validate`

**Comment by Ultraman82** [comment]
so I cant print logs with val acc? only when it eval checkpoint after each epoch?

**Comment by dreamerlin** [comment]
> so I cant print logs with val acc? only when it eval checkpoint after each epoch?

When you use `--validate`, it will call `EvalHook` to calculate the **val** acc on validation dataset using the current model rather than loading the checkpoint. And The log you show with high acc is calculated on the training dataset, that is **train acc**, which is obviously high since the model has 'seen' there data. : )

**Comment by Ultraman82** [comment]
Thanks guys!!! Love you! ;)

**Comment by dreamerlin** [comment]
> Thanks guys!!! Love you! ;)

If you feel we have help you, give us a STAR! : )

**Comment by Ultraman82** [comment]
I DID!

**Comment by dreamerlin** [comment]
> I DID!

Thanks !
